File: javascript/prime-factors/prime-factors.js
export function primeFactors(number) {
  const factors = [];
  let divisor = 2;

  while (number > 1) {
    while (number % divisor == 0) {
      factors.push(divisor);
      number /= divisor;
    }
    divisor += 1;
  }

  return factors;
}

File: ruby/prime-factors/prime_factors.rb
class PrimeFactors
  def self.for(number)
    factors = []
    divisor = 2

    while number > 1
      if (number % divisor).zero?
        number /= divisor
        factors << divisor
      else
        divisor += 1
      end
    end

    factors
  end
end
